packages/flutter_tools/lib/src/base/context.dart

```diff
@@ -107,6 +107,17 @@ class AppContext {
 
   /// Gets the value associated with the specified [type], or `null` if no
   /// such value has been associated.
+  T get<T>() {
+    dynamic value = _generateIfNecessary(T, _overrides);
+    if (value == null && _parent != null) {
+      value = _parent.get<T>();
+    }
+    return _unboxNull(value ?? _generateIfNecessary(T, _fallbacks)) as T;
```

Comment:
You can avoid the `as` here by saying:

```dart
T result = _unboxNull(value ?? _generateIfNecessary(T, _fallbacks));
return result;
```

(you can do the same above in `AppContext get context`)